[docs] Update Google authentication guide for Credential Manager and Nitro Google Sign-In (#46482)
expo/expo: docs/pages/guides/google-authentication.mdx

---
title: Using Google authentication
description: A guide on using react-native-nitro-google-signin (Android Credential Manager) or @react-native-google-signin/google-signin to integrate Google authentication in your Expo project.
---

import { BoxLink } from '~/ui/components/BoxLink';
import { Prerequisites, Requirement } from '~/ui/components/Prerequisites';

You can integrate Google authentication in your Expo app using either of the following libraries:

> - [`react-native-nitro-google-signin`](https://react-native-nitro-google-signin.github.io): Uses modern native APIs — including **Android Credential Manager**.

Reviewer: **[nit]** Avoid em dashes; the style guide prefers commas or separate sentences for phrase splitting (`expo-docs-style-guide.md#punctuation`).

```suggestion
- [`react-native-nitro-google-signin`](https://react-native-nitro-google-signin.github.io): Uses modern native APIs, including **Android Credential Manager**.
```

> - [`@react-native-google-signin/google-signin`](https://github.com/react-native-google-signin/google-signin): A widely used alternative implementation. Note that **Android Credential Manager APIs are available as part of their paid offering**.

Reviewer: ```suggestion
- [`@react-native-google-signin/google-signin`](https://github.com/react-native-google-signin/google-signin): A widely used library. Note that **Android Credential Manager APIs are available as part of their paid offering**.
```
I don't it is alternate, it's quite an old library. I'd suggest omitting it. I also think highlighting one part here in the introduction and basing it on Android Credential Manager is a bit too much.

the purpose of this guide is to provide a gentle introduction. To list the differences between the two, there can be a separate section that includes other differences not only the Android Credential Manager.  The introduction should stick to an introduction to both libraries and their purpose, summarized.

I'd suggest rewriting the introduction and leave the paid vs free features out of the introduction.

Both libraries provide native sign-in buttons and support authenticating the user (and obtaining authorization to use Google APIs). Because they require custom native code, you'll need to use a [config plugin](/config-plugins/introduction/) in the [app config](/versions/latest/config/app/) and build a development build.

> > The legacy Google Sign-In SDK for Android (part of `com.google.android.gms:play-services-auth`) is deprecated and Google recommends migrating to **Android Credential Manager**. See [About the migration from legacy Google Sign-In](https://developer.android.com/identity/sign-in/legacy-gsi-migration).

Reviewer: **[suggestion]** This is a deprecation message, and the callouts convention maps warnings and deprecation messages to the `> **warning**` callout type (`expo-docs-components.md#callouts`). A plain blockquote renders as a low-attention note.

```suggestion
> **warning** The legacy Google Sign-In SDK for Android (part of `com.google.android.gms:play-services-auth`) is deprecated and Google recommends migrating to **Android Credential Manager**. See [About the migration from legacy Google Sign-In](https://developer.android.com/identity/sign-in/legacy-gsi-migration).
```

---
title: Using Google authentication

> description: A guide on using react-native-nitro-google-signin (Android Credential Manager) or @react-native-google-signin/google-signin to integrate Google authentication in your Expo project.

Reviewer: ```suggestion
description: A guide on using react-native-nitro-google-signin or @react-native-google-signin/google-signin to integrate Google authentication in your Expo project.
```

### Configure your Firebase or Google Cloud Console project

> Refer to the library documentation for a more in-depth configuration guide:
>

> > - [`react-native-nitro-google-signin`](https://react-native-nitro-google-signin.github.io)

Reviewer: ```suggestion
> - [`react-native-nitro-google-signin`](https://react-native-nitro-google-signin.github.io/docs/setup/google-cloud)
```

I think the correct link here should be this. This avoid confusion.

> > - [`@react-native-google-signin/google-signin`](https://react-native-google-signin.github.io/docs/setting-up/get-config-file)

Reviewer: **[suggestion]** Keep callouts to a single short paragraph; a bulleted list inside a callout should be moved into the regular flow (`expo-docs-components.md#callouts`). Dropping the `>` prefixes renders the same content as a normal lead-in plus list.

```suggestion
Refer to the library documentation for a more in-depth configuration guide:

- [`react-native-nitro-google-signin`](https://react-native-nitro-google-signin.github.io)
- [`@react-native-google-signin/google-signin`](https://react-native-google-signin.github.io/docs/setting-up/get-config-file)
```
